### PR TITLE
Adds path/query parameter sanization

### PR DESCRIPTION
**Pull request by andrueastman**
Fixes https://github.com/microsoft/kiota-abstractions-python/issues/245

Sanitizes types passed to the path/query object dictionaries to ensure success of various object types in the parameters. 
Adds tests to validate date,time,datetime,boolean,number and uuid types.

### CODE REVIEW

**Review by baywet — COMMENTED** [review]
I think the version in the toml file at the root also needs to be updated

**Comment by baywet** [comment]
interesting, it's not configured like [the service lib](https://github.com/microsoftgraph/msgraph-sdk-python/blob/6abd71affd2f37da21e89c3aa605fc5718eb9000/pyproject.toml#L7)

I believe it's because of a combination of [this file](https://github.com/microsoft/kiota-abstractions-python/blob/3c294832dfbf87a75edbe3e5ddb24b7a240a9050/kiota_abstractions/__init__.py#L4) and [that directive](https://github.com/microsoft/kiota-abstractions-python/blob/3c294832dfbf87a75edbe3e5ddb24b7a240a9050/pyproject.toml#L24) but maybe @samwelkanda can confirm?
If so, it'd be interesting to replicate in other python repos, for consistency and efficiency.

File: kiota_abstractions/request_information.py [hunk 1]
```diff
@@ -291,17 +291,26 @@ def _set_content_and_content_type_header(
...
+        elif isinstance(value, list) and all(isinstance(x, Enum) for x in value):
+            sanitized_value = ','.join([x.value for x in value])
+        elif isinstance(value, UUID):
+            sanitized_value = str(value)
+        elif isinstance(value, datetime):
+            sanitized_value = value
+        elif isinstance(value, date):
+            sanitized_value = str(value)
+        elif isinstance(value, time):
+            sanitized_value = str(value)
```
Comment (musale) [inline kiota_abstractions/request_information.py:312]:
```suggestion
        elif any([isinstance(value, UUID), isinstance(value, date), isinstance(value, time)]:)
            sanitized_value = str(value)
        elif isinstance(value, datetime):
            sanitized_value = value
```
Collapse the `str`ingers.

**Comment by samwelkanda** [comment]
@baywet @andrueastman We do not need to update the `pyproject.toml` file for now as the core libraries use `flit` to publish packages to PyPI which only requires the VERSION variable to be updated. I agree that we should update the workflows to align with the setup in service libraries that use Python’s built-in `build` tool. @shemogumbe are you able to take this up?

**Comment by pjmagee** [comment]
Aweomse stuff

**Comment by shemogumbe** [comment]
Do we go with python build? and align with service  libraries?

**Pull request merged**